File: CinemaProject/Model/Request/ReservationRequestV1.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Model.Request
{
    public class ReservationRequestV1
    {
        public int ReservationID { get; set; }
        public string Names { get; set; }
        public string LastNames { get; set; }
        public DateTime BirthDate { get; set; }
        public string Gender { get; set; }
        public string DNI { get; set; }
        public string Email { get; set; }
        public int CinemaRoomID { get; set; }
        public int MovieID { get; set; }
        public bool Enabled { get; set; }
        public DateTime? CreatedDate { get; set; }
    }
}

File: CinemaProject/CinemaWeb/ClientApp/src/app/models/request/reservationRequestV1.ts
import { CinemaRoomResponseV1 } from "../response/cinemaRoomResponseV1";
import { MovieResponseV1 } from "../response/movieResponseV1";

export class ReservationRequestV1 {
    reservationID!: number;
    names!: string;
    lastNames!: string;
    birthDate!: Date;
    gender!: string;
    dni!: string;
    email!: string;
    cinemaRoomID!: number;
    cinemaRoom!: CinemaRoomResponseV1;
    movieID!: number;
    movie!: MovieResponseV1;
    enabled!: boolean;
    createdDate!: Date | null;
    updateDate!: Date | null;
  }